### PR TITLE
Added require-modules-from-pmr policy

### PR DESCRIPTION
**Pull request by rberlind**
I added the require-modules-from-pmr.sentinel policy which uses a Sentinel mock in contrast to the current tests that use static json mocks.  Anthony Davanzo had suggested I add this here.

The policy requires that all modules loaded by the root module come from the [Private Module Registry](https://www.terraform.io/docs/enterprise/registry/index.html) that belongs to the organization.

In addition to the policy itself, there is also a template file so that the actual policy imported into the user's organization will have their organization name.

There is also a new require-modules-from-pmr directory under the test directory with fail.json, pass.json, and corresponding sentinel mock files that they use.

I modified main.tf to create my policy from the template file.

The Sentinel mocks are based on this one: https://www.terraform.io/docs/enterprise/sentinel/import/mock-tfconfig.html#mock-file-contents

I tested `sentinel test` to make sure all tests still passed.

I also created a TFE workspace in an organization and then applied against main.tf. That worked.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -183,3 +184,20 @@ resource "tfe_sentinel_policy" "gcp-restrict-machine-type" {
   policy       = "${file("./gcp-restrict-machine-type.sentinel")}"
   enforce_mode = "soft-mandatory"
 }
+
+# Policy that requires modules to come from Private Module Registry
+data "template_file" "require-modules-from-pmr" {
+  template = "${file("./require-modules-from-pmr.template")}"
+
+  vars {
+    organization = "${var.tfe_organization}"
```
Comment (nfagerlund) [inline main.tf]:
```suggestion
    organization = "${var.tfe_organization}"
    hostname = "${var.tfe_hostname}"
```

File: require-modules-from-pmr.template [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import "tfconfig"
+import "strings"
+
+# Require all modules directly under root module
+# to come from TFE private module registry (PMR)
+require_modules_from_pmr = rule { 
+  all tfconfig.modules as _, m {
+    print("source: ", m.source) and
+    strings.has_prefix(m.source, "app.terraform.io/${organization}")
```
Comment (nfagerlund) [inline require-modules-from-pmr.template]:
```suggestion
    strings.has_prefix(m.source, "${hostname}/${organization}")
```

File: require-modules-from-pmr.sentinel [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import "tfconfig"
+import "strings"
+
```
Comment (nfagerlund) [inline require-modules-from-pmr.sentinel]:
```suggestion

# NOTE: This policy hardcodes an organization name, and is only included for testing purposes.
# For real-life use, load the policy content from require-modules-from-pmr.template using
# template_file.

```

File: main.tf [hunk 4]
```diff
@@ -183,3 +184,20 @@ resource "tfe_sentinel_policy" "gcp-restrict-machine-type" {
...
+  }
+}
+
+resource "tfe_sentinel_policy" "require-modules-from-pmr" {
+  name         = "require-modules-from-pmr"
+  description  = "Require all modules to come from the Private Module Registy of the current org"
+  organization = "${var.tfe_organization}"
+  policy       = "${data.template_file.require-modules-from-pmr.rendered}"
+  enforce_mode = "hard-mandatory"
```
Comment (nfagerlund) [inline main.tf]: Do you think maybe `soft-mandatory` might be more realistic, as a safety valve for testing modules that are in review and haven't been approved into the private registry yet?

**Comment by rberlind** [comment]
I'm going to work on this today, @nfagerlund

File: require-modules-from-pmr.template (hunk 2 above)
Comment (rberlind) [reply require-modules-from-pmr.template]: good idea.

File: require-modules-from-pmr.sentinel (hunk 3 above)
Comment (rberlind) [reply require-modules-from-pmr.sentinel]: Actually, I had not intended to include the require-modules-from-pmr.sentinel file.

Comment (rberlind) [reply require-modules-from-pmr.sentinel]: However, I might need it for the simulator

Comment (rberlind) [reply require-modules-from-pmr.sentinel]: I'm going to see if I can write out the policy from the template locally.

Comment (rberlind) [reply require-modules-from-pmr.sentinel]: Problem is that the file from template would be written to the Terraform worker container on the TFE server, not to user's laptop from which they would run `sentinel test`

**Comment by rberlind** [comment]
Hi @nfagerlund : I made the suggested changes and tested that everything worked both when running `sentinel test` and when testing the actual policy that was created in TFE.  I tested the latter with a workspace that first specified my org's PMR and then with modified TF code that used the public module registry.  The first test passed and the second failed as desired.

Note that I removed "template" from the name of the policy file.

I then added comments indicating that "app.terraform.io/OurOrganization" should not be changed since that is what the tfconfig mocks user.

**Comment by rberlind** [comment]
Thanks @nfagerlund.  I appreciate the improvements you suggested.  Glad to have contributed to this projec.t

**Pull request merged**